### PR TITLE
gh-116000: Make ``Tools/cases_generator/optimizer_generator.py`` work without any arguments

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-116000

### CODE REVIEW

File: Tools/cases_generator/optimizer_generator.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from stack import Stack, SizeMismatch, UNUSED
 
 DEFAULT_OUTPUT = ROOT / "Python/optimizer_cases.c.h"
-DEFAULT_ABSTRACT_INPUT = ROOT / "Python/optimizer_bytecodes.c"
+DEFAULT_ABSTRACT_INPUT = (ROOT / "Python/optimizer_bytecodes.c").absolute().as_posix()
```
Comment (Eclips4) [inline Tools/cases_generator/optimizer_generator.py:30]: This change is needed because of `parser/prettify_filename` which works only with an `str` type.

**Review by gvanrossum — APPROVED** [review]
Thanks! Though honestly, I wish we weren't using pathlib at all here -- it's been nothing but trouble.

**Comment by Eclips4** [comment]
Some kind of proof that this PR works:

Run without arguments:
```python
eclips4@HOME-PC:~/CLionProjects/cpython/Tools/cases_generator$ python3.11 optimizer_generator.py 
eclips4@HOME-PC:~/CLionProjects/cpython/Tools/cases_generator$ 
```

Run with explicit arguments:
```python
eclips4@HOME-PC:~/CLionProjects/cpython/Tools/cases_generator$ python3.11 optimizer_generator.py ../../Python/optimizer_bytecodes.c ../../Python/bytecodes.c
eclips4@HOME-PC:~/CLionProjects/cpython/Tools/cases_generator$ 
```

**Comment by gvanrossum** [comment]
Would be even more convincing if after the default run `git diff` showed no changes. But I turned on auto-merge.

**Comment by Eclips4** [comment]
> Would be even more convincing if after the default r

Sure!

Run without args:
```bash
eclips4@HOME-PC:~/CLionProjects/cpython$ python3.11 Tools/cases_generator/optimizer_generator.py 
eclips4@HOME-PC:~/CLionProjects/cpython$ git diff
eclips4@HOME-PC:~/CLionProjects/cpython$
```

Run with args:
```bash
eclips4@HOME-PC:~/CLionProjects/cpython$ python3.11 Tools/cases_generator/optimizer_generator.py Python/optimizer_bytecodes.c Pyth
on/bytecodes.c 
eclips4@HOME-PC:~/CLionProjects/cpython$ git diff
eclips4@HOME-PC:~/CLionProjects/cpython$
```

**Pull request merged**